chia/full_node/full_node.py

```diff
@@ -437,27 +446,25 @@ async def short_sync_batch(self, peer: ws.WSChiaConnection, start_height: uint32
                 if not response:
                     raise ValueError(f"Error short batch syncing, invalid/no response for {height}-{end_height}")
                 async with self._blockchain_lock_high_priority:
-                    success, advanced_peak, fork_height, coin_changes = await self.receive_block_batch(
-                        response.blocks, peer, None
-                    )
+                    state_change_summary: Optional[StateChangeSummary]
+                    success, state_change_summary = await self.receive_block_batch(response.blocks, peer, None)
                     if not success:
                         raise ValueError(f"Error short batch syncing, failed to validate blocks {height}-{end_height}")
-                    if advanced_peak:
-                        peak = self.blockchain.get_peak()
+                    if state_change_summary is not None:
                         try:
                             peak_fb: Optional[FullBlock] = await self.blockchain.get_full_peak()
-                            assert peak is not None and peak_fb is not None and fork_height is not None
-                            mempool_new_peak_result, fns_peak_result = await self.peak_post_processing(
-                                peak_fb, peak, fork_height, peer, coin_changes[0]
-                            )
-                            await self.peak_post_processing_2(
-                                peak_fb, peak, fork_height, peer, coin_changes, mempool_new_peak_result, fns_peak_result
+                            assert peak_fb is not None
+                            ppp_result: PeakPostProcessingResult = await self.peak_post_processing(
+                                peak_fb,
+                                state_change_summary,
+                                peer,
                             )
+                            await self.peak_post_processing_2(peak_fb, peer, state_change_summary, ppp_result)
                         except asyncio.CancelledError:
```

Comment:
I will investigate. Post processing 1 is very important because if it's not called, we might put the node in an invalid state (since there is a memory cache of the latest slots, and the mempool, that need to get updated).